controllers/datadogagent/merger/network_policy.go

```diff
@@ -0,0 +1,203 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package merger
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-operator/apis/datadoghq/common"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/component"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/dependencies"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object"
+	"github.com/DataDog/datadog-operator/pkg/kubernetes"
+
+	netv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// NetworkPolicyManager is used to manage network policy resources.
+type NetworkPolicyManager interface {
+	AddKubernetesNetworkPolicy(name, namespace string, podSelector metav1.LabelSelector, policyTypes []netv1.PolicyType, ingress []netv1.NetworkPolicyIngressRule, egress []netv1.NetworkPolicyEgressRule) error
+	BuildKubernetesNetworkPolicy(dda metav1.Object, componentName v2alpha1.ComponentName) error
+}
+
+// NewNetworkPolicyManager returns a new NetworkPolicyManager instance
+func NewNetworkPolicyManager(store dependencies.StoreClient) NetworkPolicyManager {
+	manager := &networkPolicyManagerImpl{
+		store: store,
+	}
+	return manager
+}
+
+// networkPolicyManagerImpl is used to manage network policy resources.
+type networkPolicyManagerImpl struct {
+	store dependencies.StoreClient
+}
+
+// AddKubernetesNetworkPolicy creates or updates a kubernetes network policy
+func (m *networkPolicyManagerImpl) AddKubernetesNetworkPolicy(name, namespace string, podSelector metav1.LabelSelector, policyTypes []netv1.PolicyType, ingress []netv1.NetworkPolicyIngressRule, egress []netv1.NetworkPolicyEgressRule) error {
+	obj, _ := m.store.GetOrCreate(kubernetes.NetworkPoliciesKind, namespace, name)
+	policy, ok := obj.(*netv1.NetworkPolicy)
+	if !ok {
+		return fmt.Errorf("unable to get from the store the NetworkPolicy %s", name)
+	}
+
+	policy.Spec.PodSelector = podSelector
+	policy.Spec.PolicyTypes = append(policy.Spec.PolicyTypes, policyTypes...)
+	policy.Spec.Ingress = append(policy.Spec.Ingress, ingress...)
+	policy.Spec.Egress = append(policy.Spec.Egress, egress...)
+	return m.store.AddOrUpdate(kubernetes.NetworkPoliciesKind, policy)
+}
+
+// BuildAgentKubernetesNetworkPolicy creates the base node agent kubernetes network policy
+func (m *networkPolicyManagerImpl) BuildKubernetesNetworkPolicy(dda metav1.Object, componentName v2alpha1.ComponentName) error {
```

Comment:
See my comment about `BuildAgentLocalService` in the `service.go` file.
I think the same applies to this function.